funfact/optimizable.py

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+from funfact.backend import active_backend as ab
+from typing import Any
+
+
+class Optimizable:
+
+    def __init__(self, mode: bool) -> None:
+        global _all_optimizable
+        self._prev_mode = _all_optimizable
+        _all_optimizable = mode
+        if 'torch' in ab.__repr__().lower():
+            ab.set_grad_enabled(mode)
+
+    def __enter__(self) -> None:
+        pass
```

Comment:
Having an empty `__enter__` is a bit suspicious to me.

Reply:
This is actually quite similar to the PyTorch implementation: https://pytorch.org/docs/stable/_modules/torch/autograd/grad_mode.html#set_grad_enabled

Do you think it's better to move the `__init__` code to `__enter__`?

Reply:
Given that this context manager is relatively lightweight, can we simply use [contextlib](https://docs.python.org/3/library/contextlib.html) to create it?

Reply:
I mean the `@contextlib.contextmanager` decorator :smile: